### PR TITLE
Added initial set of tests for Percy along with config file

### PR DESCRIPTION
**Pull request by SherBhachuOnFido**
# Problem
- We did not have any way of tracking or seeing visual regressions, where current tests may have still been passing.

# Solution
- Add Percy visual regression testing across a subset of existing tests.

## Checklist

_put `n/a` if item is not relevant to PR changes_

- [x] Has the CHANGELOG been updated?
- [ ] Has the README been updated?
- [ ] Has the CONTRIBUTING doc been updated?
- [ ] Has the RELEASE_GUIDELINES been updated?
- [ ] Has the MIGRATION doc been updated for any MAJOR breaking changes?
- [ ] Has the MIGRATION doc been updated for any MINOR breaking changes, including any translation strings or keys changes?
- [x] Have any new automated tests been implemented or the existing ones changed?
- [ ] Have any new manual tests been written down or the existing ones changed?
- [ ] Have any new strings been translated or the existing ones changed?

### CODE REVIEW

File: .github/workflows/percy.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: Percy Visual Regression Tests
+
+on:
+  workflow_dispatch:
```
Comment (stefaniacardenas) [inline .github/workflows/percy.yml:4]: Are we planning to only trigger this manually or shall we set up a cron job?

File: .github/workflows/percy.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Install Dependencies
+        run: npm ci
```
Comment (stefaniacardenas) [inline .github/workflows/percy.yml:43]: `npm ci` is currently failing in our builds, we need to look into it to understand why. For now you can use `npm install`. This means that lines 24 to 32 can be deleted for now

File: licenses.json [hunk 3]
```diff
@@ -9119,7 +9121,7 @@
     "publisher": "Web SDK Customer Support",
     "email": "web-sdk@onfido.com",
     "url": "https://github.com/onfido",
-    "path": "onfido-sdk-ui",
+    "path": "/Users/sher.bhachu/Documents/Repo/Public/onfido-sdk-ui",
```
Comment (stefaniacardenas) [inline licenses.json:9124]: This script is a bit annoying and it inserts your local repo path. You have to manually delete it.

File: package.json [hunk 4]
```diff
@@ -31,6 +31,7 @@
     "test:integration": "jest --verbose --testPathPattern=__integrations__",
     "test:unit": "jest --coverage --verbose --testPathPattern=__tests__",
     "test:ui": "babel-node test/main.js",
+    "test:percy": "CONFIG_FILE=percy.json PERCY=true percy exec npm run test:ui",
```
Comment (stefaniacardenas) [inline package.json:34]: Where are you using this flag `PERCY=true`?

File: test/main.js [hunk 5]
```diff
@@ -178,7 +178,7 @@ const createMocha = (driver, testCase) => {
     )
   })
   mocha.suite.beforeEach('Set retry', function () {
-    this.currentTest.retries(2)
+    this.currentTest.retries(1)
```
Comment (stefaniacardenas) [inline test/main.js:181]: Is this going to speed up tests feedback?

File: test/specs/percy.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  // Note: The SDK works also with language tags that do not include region (e.g. 'en', 'es')
+  // We are passing the region here so we can fetch the right json file path (e.g. `en_US/en_US.json`).
+  supportedLanguages.forEach((lang) => {
+    //welcomeScenarios(lang)
+  })
+  // PoA is only available in en
+  proofOfAddressScenarios()
+  //accessibilityScenarios()
+  //hostAppHistoryScenarios()
+  //userConsentScenarios()
```
Comment (stefaniacardenas) [inline test/specs/percy.js:37]: Why are some scenarios commented out? If we are not planning to test those now we can just delete them and bring them back when needed

File: test/specs/scenarios/face.js [hunk 7]
```diff
@@ -44,6 +45,12 @@ export const faceScenarios = (lang) => {
 
     const copy = basePage.copy(lang)
 
+    async function takePercySnapshotWithoutOverlay(driver, text) {
```
Comment (stefaniacardenas) [inline test/specs/scenarios/face.js:48]: Why is this needed?

File: package.json (hunk 4 above)
Comment (stefaniacardenas) [reply package.json:34]: Found it

File: .github/workflows/percy.yml (hunk 1 above)
Comment (SherBhachuOnFido) [reply .github/workflows/percy.yml:4]: Manually for the moment, until we come up with a proper CI strategy, that would include things like general automation coverage and such.

File: .github/workflows/percy.yml (hunk 2 above)
Comment (SherBhachuOnFido) [reply .github/workflows/percy.yml:43]: Noted. I shall remove.

File: licenses.json (hunk 3 above)
Comment (SherBhachuOnFido) [reply licenses.json:9124]: Thanks, I didnt even notice that!

File: test/main.js (hunk 5 above)
Comment (SherBhachuOnFido) [reply test/main.js:181]: Well, I noticed that we don't need to try 2 additional times. Our main issues on retires at the moment is this general 'Async....' errors.

File: test/specs/percy.js (hunk 6 above)
Comment (SherBhachuOnFido) [reply test/specs/percy.js:37]: Oh, this was my mistake, I thought I had commented them out :D

File: test/specs/scenarios/face.js (hunk 7 above)
Comment (SherBhachuOnFido) [reply test/specs/scenarios/face.js:48]: So, we have some tests where there is a 'video' overlay placed by the tests, and because of the way the screenshots are taken by Percy they obstruct the view. This method will take the screenshot and ignore the overlay...

Comment (SherBhachuOnFido) [reply test/specs/scenarios/face.js:48]: Its so we DONT see something like this captured...
Screenshot 2021-05-24 at 18 14 24
But we see...
Screenshot 2021-05-24 at 18 15 48

File: test/specs/scenarios/face.js [hunk 8]
```diff
@@ -332,20 +369,50 @@ export const faceScenarios = (lang) => {
...
       confirm.checkLogoIsHidden()
+      await takePercySnapshot(
+        driver,
+        `Verify Check selfie video screen does not have onfido logo ${lang}`
+      )
       confirm.clickConfirmButton()
       verificationComplete.checkLogoIsHidden()
+      await takePercySnapshot(
+        driver,
+        `Verify Verification complete screen does not have onfido logo ${lang}`
```
Comment (kopijunkie) [inline test/specs/scenarios/face.js:414]: Wonder if we really need the "Verify ..." bit at the start of each test. It also just happens to sound a bit weird here because the name of the screen itself has "verification" also.
```suggestion
        `Verify Verification complete screen does not have Onfido logo ${lang}`
```

File: test/specs/scenarios/face.js [hunk 9]
```diff
@@ -332,20 +369,50 @@ export const faceScenarios = (lang) => {
...
+      camera.nextChallengeButton().click()
+      await takePercySnapshotWithoutOverlay(
+        driver,
+        `Vocal challenge is given ${lang}`
+      )
+      camera.stopButton().click()
       confirm.checkLogoIsHidden()
+      await takePercySnapshot(
+        driver,
+        `Verify Check selfie video screen does not have onfido logo ${lang}`
```
Comment (kopijunkie) [inline test/specs/scenarios/face.js:408]:
```suggestion
        `Verify Check selfie video screen does not have Onfido logo ${lang}`
```

**Review by kopijunkie — CHANGES_REQUESTED** [review]
Looks good. Just minor test description edits.

File: licenses.json [hunk 10]
```diff
@@ -945,14 +945,14 @@
     "path": "node_modules/@onfido/castor-icons",
     "licenseFile": "node_modules/@onfido/castor-icons/README.md"
   },
-  "@onfido/castor-react@1.2.0": {
+  "@onfido/castor-react@1.1.2": {
```
[the remark this answers is not in the document]
Comment (SherBhachuOnFido) [reply licenses.json:948]: What do you think I should do?

File: test/specs/scenarios/document.js [hunk 11]
```diff
@@ -40,6 +41,10 @@ export const documentScenarios = async (lang) => {
 
       const runThroughPassportUploadFlow = async () => {
         documentUpload.clickUploadButton()
+        await takePercySnapshot(
+          driver,
+          `Verify Upload passport photo page ${lang}`
```
[the remark this answers is not in the document]
Comment (SherBhachuOnFido) [reply test/specs/scenarios/document.js:46]: Ah, ok, I'll go through and make that change to all the percy snapshot calls.

File: licenses.json (hunk 10 above)
[the remark this answers is not in the document]
Comment (kopijunkie) [reply licenses.json]: I'd try re-running `npm install && npm run license-checker` locally to see if the versions are still different in the generated file.

[the remark this answers is not in the document]
Comment (SherBhachuOnFido) [reply licenses.json]: OK, noted. I'll try that.

File: .github/workflows/percy.yml (hunk 2 above)
Comment (stefaniacardenas) [reply .github/workflows/percy.yml]: yes, it does

**Review by stefaniacardenas — APPROVED** [review]

File: .github/workflows/percy.yml (hunk 2 above)
Comment (SherBhachuOnFido) [reply .github/workflows/percy.yml]: Done.
